### ISSUE TITLE
Match fill arms postfix

### ISSUE THREAD

**Issue by JosephTLyons** [rust-analyzer/rust-analyzer#11409]
If you have

```rust
let a = Some(1);
```

when you type `a.match` and complete it, you get

```rust
match a {
    Some(_) => {},
    None => {},
}
```

If you have a custom `enum`

```rust
enum State {
    Blocked,
    Running,
    Stopped,
}
```

```
let state = State::Blocked;
```

typing `state.match` and completing it gives

```rust
match state {
    _ => {},
}
```
As a note, RA has an assist to automatically fill arms:

![SCR-20220204-39u](https://user-images.githubusercontent.com/19867440/152488269-1e679cec-af67-403d-8c69-5a50d9991e26.png)

**Comment by Veykril** [comment]
The reason why we aren't autofilling arms on completion in the general is that a user may not want to always match all variants of an enum(option is special because there are 2 variants and you have to match with 2 arms for the match to be useful)

**Comment by Nadrieril** [comment]
Could there be a threshold, maybe fill all of them if there are only 2-3 cases since it's easy to remove 1-2 cases?

**Comment by Veykril** [comment]
Yes, I think generalizing this for matches of 1 and 2 arms makes sense, 3 I would say no to as in that case you might want to go back to 2 arms, the interesting match and a `_` fallback. Whereas with 2 arms you'd merely replace the uninteresting one with `_`.

**Comment by jonathancyu** [comment]
Is it viable to create a .matchall postfix snippet? I understand the reasoning, however I find myself expanding .match then using the fill match arms code action that I think having a snippet that fills all arms is warranted.

**Comment by Nadrieril** [comment]
This could alternatively be helped by a ["split this pattern into subpatterns"](https://github.com/rust-lang/rust-analyzer/issues/8365) code action: step one generates a match with `_`, step two expands it into all the variants.
